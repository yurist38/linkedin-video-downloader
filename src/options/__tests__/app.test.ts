import { VueConstructor } from 'vue';
import { shallowMount, mount, createLocalVue, Wrapper } from '@vue/test-utils';
import App from '../app.vue';

describe('options/app', () => {
  let localVue: VueConstructor;
  let wrapper: Wrapper<App>;

  beforeEach(() => {
    localVue = createLocalVue();
  });

  it('Should render correctly', () => {
    process.env.APPZI_TOKEN = 'test-value';
    process.env.APPZI_ID = 'test-value';

    wrapper = mount(App, {
      localVue,
      stubs: ['b-button']
    });

    expect(wrapper.element).toMatchSnapshot();

    const script = document.body.querySelector('script[src*="test-value"]') as HTMLScriptElement;
    (script.onload as Function)();
    expect(wrapper.vm.$data.appziLoaded).toEqual(true);
  });

  it('Should throw an error when appzi token is not provided', () => {
    process.env.APPZI_TOKEN = '';

    expect(() => shallowMount(App, {
      localVue,
      stubs: ['b-button']
    })).toThrowError('Appzi credentials are not provided...');
  });

  it('Should throw an error when appzi id is not provided', () => {
    process.env.APPZI_ID = '';

    expect(() => shallowMount(App, {
      localVue,
      stubs: ['b-button']
    })).toThrowError('Appzi credentials are not provided...');
  });
});
