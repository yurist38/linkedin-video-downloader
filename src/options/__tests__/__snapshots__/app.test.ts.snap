// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`options/app Should render correctly 1`] = `
<section
  class="section"
>
  <div
    class="container"
  >
    <div
      class="centered"
    >
      <img
        src="../../assets/icons/icon128.png"
      />
       
      <h1
        class="title"
      >
        LinkedIn Video Downloader
      </h1>
    </div>
     
    <div
      class="notification info-text"
    >
      <p>
        
        Thank you for using LinkedIn Video Downloader!
        How do you feel about it? Maybe you have an idea for improvement or want to report a bug?
        Please consider sending us a feedback message...
      
      </p>
    </div>
     
    <div
      class="centered"
    >
      <b-button-stub
        class="feedback-btn is-outlined"
        data-az-l="test-value"
        disabled="true"
        id="feedback-btn"
        rounded=""
        size="is-large"
        type="is-primary"
      >
        
        Send feedback
      
      </b-button-stub>
    </div>
  </div>
</section>
`;
